lib/executor/instantiate/component/component_canon.cpp

```diff
@@ -13,260 +13,79 @@ namespace Executor {
-        Rets[RI++].emplace<std::string>(std::string(Str.begin(), Str.end()));
-        break;
-      }
-      default: {
-        Rets[RI++].emplace<ValVariant>(ResultList[TakeI++].first);
-        break;
-      }
-      }
-    }
-
-    return {};
-  }
-
-private:
-  Executor *Exec;
-  Runtime::Instance::FunctionInstance *LowerFunc;
-  Runtime::Instance::MemoryInstance *Memory;
-  Runtime::Instance::FunctionInstance *Realloc;
-};
-
-std::unique_ptr<Runtime::Instance::Component::FunctionInstance>
-Executor::lifting(Runtime::Instance::ComponentInstance &Comp,
-                  const AST::Component::FuncType &FuncType,
-                  Runtime::Instance::FunctionInstance *Func,
-                  Runtime::Instance::MemoryInstance *Memory,
-                  Runtime::Instance::FunctionInstance *Realloc) {
-  auto R = std::make_unique<Runtime::Instance::Component::FunctionInstance>(
-      std::make_unique<LiftTrans>(this, FuncType, Func, Memory, Realloc, Comp));
-  return R;
+  return CoreVals;
 }
 
-class LowerTrans : public Runtime::HostFunctionBase {
-public:
-  LowerTrans(Executor *E, Runtime::Instance::Component::FunctionInstance *F,
-             Runtime::Instance::MemoryInstance *M,
-             Runtime::Instance::FunctionInstance *R)
-      : Runtime::HostFunctionBase(0), Exec(E), HigherFunc(F), Memory(M),
-        Realloc(R) {
-    const auto &HigherType = HigherFunc->getFuncType();
-
-    auto &FuncType = DefType.getCompositeType().getFuncType();
-    for (auto &ParamTy : HigherType.getParamTypes()) {
-      switch (ParamTy.getCode()) {
-      case TypeCode::String:
-        FuncType.getParamTypes().push_back(TypeCode::I32);
-        FuncType.getParamTypes().push_back(TypeCode::I32);
-        break;
-      default:
-        FuncType.getParamTypes().push_back(ParamTy);
-        break;
-      }
+std::vector<std::pair<ComponentValVariant, ComponentValType>>
+Executor::convValsToComponent(
+    Span<const std::pair<ValVariant, ValType>> CoreVals,
+    Span<const ComponentValType> ValTypes,
+    Runtime::Instance::MemoryInstance *MemInst) noexcept {
+  uint32_t I = 0;
+  std::vector<std::pair<ComponentValVariant, ComponentValType>> Vals;
+  for (const auto &Type : ValTypes) {
+    switch (Type.getCode()) {
+    case ComponentTypeCode::String: {
+      auto Off = CoreVals[I++].first.get<uint32_t>();
+      auto Size = CoreVals[I++].first.get<uint32_t>();
+      auto Str = MemInst->getStringView(Off, Size);
+      Vals.emplace_back(std::string(Str), Type);
+      break;
     }
-
-    for (auto &ReturnTy : HigherType.getReturnTypes()) {
-      switch (ReturnTy.getCode()) {
-      case TypeCode::String:
-        FuncType.getReturnTypes().push_back(TypeCode::I32);
-        FuncType.getReturnTypes().push_back(TypeCode::I32);
-        break;
-      default:
-        FuncType.getReturnTypes().push_back(ReturnTy);
-        break;
-      }
+    case ComponentTypeCode::TypeIndex: {
+      // TODO: COMPONENT - Not implemented.
```

Comment:
Should we display an error message or a warning message so people know they are encountering this part?